packages/mui-joy/src/styles/styleFunctionSx.ts

```diff
@@ -0,0 +1,181 @@
+});
+
+const borderTopColor = style({
+  prop: 'borderTopColor',
+  themeKey: 'vars.palette',
+});
+
+const borderRightColor = style({
+  prop: 'borderRightColor',
+  themeKey: 'vars.palette',
+});
+
+const borderBottomColor = style({
+  prop: 'borderBottomColor',
+  themeKey: 'vars.palette',
+});
+
+const borderLeftColor = style({
+  prop: 'borderLeftColor',
+  themeKey: 'vars.palette',
+});
+
+const borderRadius = style({
+  prop: 'borderRadius',
+  themeKey: 'vars.radius',
+});
+
+const borders = compose(
+  border,
+  borderTop,
+  borderRight,
+  borderBottom,
+  borderLeft,
+  borderColor,
+  borderTopColor,
+  borderRightColor,
+  borderBottomColor,
+  borderLeftColor,
+  borderRadius,
+);
+
+const color = style({
+  prop: 'color',
+  themeKey: 'vars.palette',
+});
+
+const bgcolor = style({
+  prop: 'bgcolor',
+  cssProperty: 'backgroundColor',
+  themeKey: 'vars.palette',
+});
+
+const backgroundColor = style({
+  prop: 'backgroundColor',
+  themeKey: 'vars.palette',
+});
+
+const palette = compose(color, bgcolor, backgroundColor);
+
+const boxShadow = style({
+  prop: 'boxShadow',
+  themeKey: 'vars.shadow',
+});
+
+export const fontFamily = style({
+  prop: 'fontFamily',
+  themeKey: 'vars.fontFamily',
+});
+
+export const fontSize = style({
+  prop: 'fontSize',
+  themeKey: 'vars.fontSize',
+});
+
+export const fontWeight = style({
+  prop: 'fontWeight',
+  themeKey: 'vars.fontWeight',
+});
+
+export const letterSpacing = style({
+  prop: 'letterSpacing',
+  themeKey: 'vars.letterSpacing',
+});
+
+export const lineHeight = style({
+  prop: 'lineHeight',
+  themeKey: 'vars.lineHeight',
+});
+
+const typography = compose(
+  typographyVariant,
+  fontFamily,
+  fontSize,
+  fontWeight,
+  letterSpacing,
+  lineHeight,
+);
+
+const filterPropsMapping = {
+  borders: borders.filterProps,
+  display: display.filterProps,
+  flexbox: flexbox.filterProps,
+  grid: grid.filterProps,
+  positions: positions.filterProps,
+  palette: palette.filterProps,
+  boxShadow: boxShadow.filterProps,
+  sizing: sizing.filterProps,
+  spacing: spacing.filterProps,
+  typography: typography.filterProps,
+};
+
+const styleFunctionMapping = {
+  borders,
+  display,
+  flexbox,
+  grid,
+  positions,
+  palette,
+  boxShadow,
+  sizing,
+  spacing,
+  typography,
+};
+
+const propToStyleFunction = Object.keys(filterPropsMapping).reduce((acc, styleFnName) => {
+  // @ts-ignore private function
+  filterPropsMapping[styleFnName].forEach((propName) => {
+    // @ts-ignore private function
+    acc[propName] = styleFunctionMapping[styleFnName];
+  });
+
+  return acc;
+}, {});
+
+function getThemeValue(prop: string, value: string | number | object, theme: object) {
+  const inputProps = {
+    [prop]: value,
+    theme,
+  };
+
+  // @ts-ignore private function
+  const styleFunction = propToStyleFunction[prop];
+  return (styleFunction ? styleFunction(inputProps) : { [prop]: value }) as object;
+}
+
+const styleFunctionSx = unstable_createStyleFunctionSx({
```

Comment:
I think the only thing we need to pass is the `filterPropsMapping` and the `styleFunctionMapping`, based on that we can create the methods inside the `createStyleFunctionSx`. It is even logical if you think about it, the things needed to provide for the `sx` prop to work is, which props you want to support and how are those mapped to the them. These methods that you are sending are just internal functions about how we implemented the `styleFunctionSx`. We may change the internal implementaiton in the future, and those may not even exist.

Reply:
Updated! looks much better. Please review again.